torch/csrc/utils/tensor_new.cpp

```diff
@@ -279,10 +293,21 @@ Tensor internal_new_from_data(
     maybe_initialize_cuda(device);
     return tensor.to(device, inferred_scalar_type, /*non_blocking=*/false, /*copy=*/copy_numpy);
   }
+
+  std::vector<PyArrayObject*> array_ptr_storage;
+  if (torch::utils::extract_ndarrays(data, sizes, array_ptr_storage)) {
```

Comment:
Refactor this block so it creates the tensor and acquires the scalar type, but then code common to both this block and the next block acquires the device, release the gil, possibly initializes CUDA, calls to() on the tensor and returns it. (This will require a small edit to the next block, too.)

Reply:
Its done.